tests/e2e_suite.py

```diff
@@ -72,7 +73,17 @@ def run(args):
     success = True
     elapsed = args.test_duration
 
-    for i, test in enumerate(chosen_suite):
+    if args.filter is not None:
+        filter_re = re.compile(args.filter)
+
+        def filter_fun(x):
+            return filter_re is None or filter_re.match(x[1].__name__)
+
+        tests_to_run = filter(filter_fun, enumerate(chosen_suite))
```

Comment:
```
tests_to_run = filter(lambda x: re.compile(args.filter or '.*').match(x[1].__name__), enumerate(chosen_suite))
```
was quite tempting, but this is more readable :)

Reply:
Guess what I had before making it readable :)